packages/optimizer/lib/transformers/RewriteAmpUrls.js

```diff
@@ -49,16 +58,33 @@ class RewriteAmpUrls {
     if (!head) return;
 
     let ampUrlPrefix = params.ampUrlPrefix || AMP_CACHE_HOST;
+    let dynamicAmpUrlPrefix = params.ampUrlPrefix || AMP_CACHE_HOST;
     if (params.ampRuntimeVersion) {
-      ampUrlPrefix = appendRuntimeVersion(ampUrlPrefix, params.ampRuntimeVersion);
+      if (!params.ampUrlPrefix) {
+        ampUrlPrefix = appendRuntimeVersion(ampUrlPrefix, params.ampRuntimeVersion);
+        dynamicAmpUrlPrefix = appendRuntimeVersion(dynamicAmpUrlPrefix, params.ampRuntimeVersion);
+      } else if (params.rewriteDynamicComponents === false) {
+        dynamicAmpUrlPrefix = appendRuntimeVersion(AMP_CACHE_HOST, params.ampRuntimeVersion);
+      }
     }
 
+    // Should be kept up to date with dynamic components listed here:
+    // https://github.com/ampproject/amphtml/blob/master/spec/amp-cache-guidelines.md#guidelines-adding-a-new-cache-to-the-amp-ecosystem
+    const dynamicElements = ['amp-geo'];
+    const dynamicTemplates = [];
+
     let node = head.firstChild;
     let referenceNode = findMetaViewport(head);
 
     while (node) {
       if (node.tagName === 'script' && this._usesAmpCacheUrl(node.attribs.src)) {
-        node.attribs.src = this._replaceUrl(node.attribs.src, ampUrlPrefix);
+        const isDynamicComponent =
```

Comment:
please extract this into a separate method for readability

Reply:
Let me know if commit 3354b6a does this adequately.